features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMetaDeserializer.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+import com.google.gson.JsonParser
+
+internal class RumEventMetaDeserializer(
+    private val internalLogger: InternalLogger
+) : Deserializer<ByteArray, Any> {
+    override fun deserialize(model: ByteArray): Any? {
+        if (model.isEmpty()) return null
+
+        return try {
+            RumEventMeta.fromJson(model.toJson())
+        } catch (@Suppress("TooGenericExceptionCaught") e: NullPointerException) {
```

Comment:
> [!NOTE]
> What would be the cause of an NPE in the `fromJson`? We don't throw it ourselves, is this something that happened during your tests? If so, maybe we can prevent it from throwing instead of catching it?

Reply:
It comes from Java interop with `gson` library, because `get` call can return `null`.

Basically the list of exceptions caught is the same as we use in RUM events parsing.